pkg/tsdb/prometheus/prometheus.go

```diff
@@ -248,7 +248,7 @@ func (s *Service) parseQuery(queries []backend.DataQuery, dsInfo *DatasourceInfo
 			return nil, err
 		}
 
-		calculatedInterval := s.intervalCalculator.Calculate(queries[0].TimeRange, minInterval)
+		calculatedInterval := s.intervalCalculator.Calculate(queries[0].TimeRange, minInterval, queryModel.MaxDataPoints)
```

Comment:
NOTE: 
the value we are calculating here should be calculated already in `queries[0].Interval` ... but i understand that you want to go with small pull-request-steps :+1:

Reply:
sorry, my mistake here, please ignore this comment.

the value in `queries[0].Interval` did not "respect" `prometheus_min_step`, so it is not the correct value.